x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -595,7 +616,7 @@ export class ExecutionHandler<
     }
 
     // we fetch summarizedAlerts to filter alerts in memory as well
-    if (!isSummaryAction(action) && !action.alertsFilter) {
+    if (!isSummaryAction(action) && action.version && !action.alertsFilter) {
```

Comment:
Not sure about this, looks like here is when it was checking that is no summary action but I'm not sure what action.alertsFilter is. Anyways, IMO we should update this `shouldGetSummarizedAlerts` so it works with the new version parameter